File: Android/src/main/java/com/cisco/or/sdk/services/UsageStatisticsService.kt
package com.cisco.or.sdk.services

import android.content.Context
import com.cisco.or.sdk.types.ServiceHandler
import com.cisco.or.sdk.utils.Urls

/**
 * Gets a device's usage statistics
 */

internal object UsageStatisticsService : Service(){
    /**
     * Method responsible to get a device's usage statistics
     *
     * @param context Context of the functionality.
     * @param sdkToken User's access token.
     * @param serviceHandler Type of handler to response.
     * @author Fabiana Garcia
     * @since 24/03/2020
     */
    override fun onStart(context: Context, headerParams: ArrayList<HeaderParam>, serviceHandler: ServiceHandler) {
        val url = Urls.usageStatisticsUrl()
        val requestData = RequestData(url, headerParams, HTTP_METHOD.GET, null)
        call(context, requestData, RESPONSE_FORMATTER.JSONArray, serviceHandler)
    }
}

File: IOS/Sources/OpenRoaming/OpenRoaming/Services/UsageStatisticsService.swift
//
//  UsageStatisticsService.swift
//  OpenRoaming
//
//  Created by Fabiana Garcia on 4/27/20.
//  Copyright © 2020 Cisco. All rights reserved.
//

/**
* Gets a device's usage statistics.
*/
class UsageStatisticsService: Service {
    
    /**
     Method responsible to get a device's usage statistics.
     
     Parameter <accessToken>: User's access token.
     Parameter <serviceHandler>: Type of handler to response.
     */
    override func onStart(headerParams: Array<HeaderParam>, serviceHandler: @escaping ServiceHandler) {
        let urlString = Urls.usageStatisticsUrl()
        let requestData = RequestData(url: urlString, headerParams: headerParams, method: HTTPMethod.GET)
        call(requestData: requestData, serviceHandler: serviceHandler)
    }
}
